src/lib/ppx_version/versioned_module.ml

```diff
@@ -949,7 +989,82 @@ let convert_modbody ~loc ~version_option body =
               in
               [ all_tag_convert_guard; all_tag_convert; all_tag_versions ]
           in
-          top_tag_modules @ all_tag_modules
+          let json_tag_modules =
+            if not json_version_tag then []
+            else
+              let json_tag_versions =
+                [%stri
+                  let (json_tag_versions :
```

Comment:
This is unfortunate, it would be nice if there was a better way than hard-codingm

Reply:
This is the same approach we've used for `bin_io`.

Another way would be to record the available Vn modules, and try each one's `of_yojson` in turn. But that's less efficient than searching the assoc array.

Reply:
How else?

Reply:
I guess you're right, yea. Lgtm then